src/main/java/com/example/ludogoriesoft/lukeriaerpapi/services/ProductService.java

```diff
@@ -48,6 +50,13 @@ public ProductDTO createProduct(ProductDTO productDTO) {
         return modelMapper.map(product, ProductDTO.class);
     }
 
+    public List<ProductDTO> getProductsForSale() {
+        List<Product> products = productRepository.getAvailableProductsForSale();
+        Type listType = new TypeToken<List<ProductDTO>>() {}.getType();
+        List<ProductDTO> productDTOList = modelMapper.map(products, listType);
+        return productDTOList;
```

Comment:
you don't need to create the _**productDTOList**_ variable, you can directly return its value